auth/oidc.go

```diff
@@ -423,3 +456,50 @@ func SetURLQueryParam(strURL, name, value string) (string, error) {
 	uri.RawQuery = rawQuery.Encode()
 	return uri.String(), nil
 }
+
+func (client *OIDCClient) GetProvider() (provider *oidc.Provider) {
+	for {
+		client.mux.Lock()
+		if client.provider != nil {
+			provider = client.provider
+			client.mux.Unlock()
+			break
+		}
+		base.Infof(base.KeyAuth, "Getting provider config ready; waiting for discovery sync.")
+		client.mux.Unlock()
+		time.Sleep(100 * time.Millisecond)
+	}
+	return
+}
+
+func (client *OIDCClient) SetProvider(provider *oidc.Provider) {
+	client.mux.Lock()
+	client.provider = provider
+	client.mux.Unlock()
+}
+
+func (client *OIDCClient) SyncProviderConfig(issuer string) {
```

Comment:
Do we have an approach to test the provider sync/refresh mechanism?

I think it should be feasible to write a test provider that changes configuration and override the sync time for testing purposes.